iree/testing/benchmark_main.cc

```diff
@@ -17,9 +17,13 @@
 
 namespace iree {
 
+// Register IREE module benchmarks at runtime.
+extern "C" void RegisterModuleBenchmarks();
+
 extern "C" int main(int argc, char** argv) {
   ::benchmark::Initialize(&argc, argv);
   iree::InitializeEnvironment(&argc, &argv);
+  RegisterModuleBenchmarks();
```

Comment:
Hmmm I think it makes sense for the generic benchmark main to not deal with this. Then all users have to do some wonky function declaration. Instead, let's keep benchmark_main as a generic thing people can link in with no other setup.

So I would revert the changes in this file and write a separate main function in benchmark_module_main. If we want to share the initialization logic, let's factor out a separate function for `InitializeBenchmark` or something (although at only two lines, I'm not sure it's worth it). I think it's reasonable for a weirdo main function doing weirdo dynamic registration to have to write its own 

```suggestion
extern "C" int main(int argc, char** argv) {
  ::benchmark::Initialize(&argc, argv);
  iree::InitializeEnvironment(&argc, &argv);
```